scan/enum.go

```diff
@@ -0,0 +1,100 @@
+package scan
+
+import (
+	"go/ast"
+	"strconv"
+	"unicode"
+	"strings"
+)
+
+func enumName(comments *ast.CommentGroup, typeSpec *ast.TypeSpec) (name string, isEnum bool) {
+	if comments != nil {
+		for _, cmt := range comments.List {
+			for _, ln := range strings.Split(cmt.Text, "\n") {
+				matches := rxEnumOverride.FindStringSubmatch(ln)
+				if len(matches) > 1 {
+					name = matches[1]
+					if name == "" {
+						return typeSpec.Name.String(), true
+					}
+					return strings.TrimSpace(name), true
+				}
+			}
+		}
+	}
+	return "", false
+}
+
+func upperSnakeCase(s string) string {
```

Comment:
I see, this is great though what you're doing here.
You know one possible improvement could be to just use the value of the enum constant. Swagger allows for enums to be of any type, they can even be complex objects. Although in reality I think they are typically strings, int, uint